drivers/pinctrl/pinctrl_xmc4xxx.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2022 Schlumberger
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT infineon_xmc4xxx_pinctrl
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/dt-bindings/pinctrl/xmc4xxx-pinctrl.h>
+
+#include <xmc_gpio.h>
+
+static const struct device *const gpio_devs[] = {
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio0)),
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio1)),
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio2)),
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio3)),
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio4)),
+	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio5)),
+};
+
+static int pinctrl_configure_pin(const pinctrl_soc_pin_t pinmux)
+{
+	int port_id, pin, alt_fun, hwctrl;
+	XMC_GPIO_CONFIG_t pin_cfg = {0};
+	XMC_GPIO_PORT_t *gpio_port;
+
+	port_id = XMC4XXX_PINMUX_GET_PORT(pinmux);
+	if (port_id >= ARRAY_SIZE(gpio_devs)) {
+		return -EINVAL;
+	}
+
+	if (!device_is_ready(gpio_devs[port_id])) {
+		return -ENODEV;
+	}
```

Comment:
why is this required?

Reply:
I use it as a check to make sure gpioX has been defined in the device tree.

Reply:
but why? I still don't get the value of adding this.

Reply:
I thought you said to keep it in: 
https://github.com/zephyrproject-rtos/zephyr/pull/47581#discussion_r949131795

Probably a misunderstanding on my part. I'll remove it in the next push